compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -607,6 +615,31 @@ void generatePlatformIns(JInstruction ins) {
         }
     }
 
+    void generateObserveJIMethodCallIns(ObservabilityJMethodCall callIns, int localVarOffset) {
+        boolean isInterface = callIns.invocationType == INVOKEINTERFACE;
+        int argIndex = 0;
+        String jMethodVMSig = callIns.jMethodVMSig;
+        boolean hasBalEnvParam = jMethodVMSig.startsWith(BAL_ENV_PARAM);
+        if (hasBalEnvParam) {
+            mv.visitTypeInsn(NEW, BAL_ENV);
+            mv.visitInsn(DUP);
+            this.mv.visitVarInsn(ALOAD, localVarOffset); // load the strand
+            // load the current Module
+            mv.visitFieldInsn(GETSTATIC, this.moduleInitClass, CURRENT_MODULE_VAR_NAME, GET_MODULE);
+            mv.visitMethodInsn(INVOKESPECIAL, BAL_ENV, JVM_INIT_METHOD,
+                    INIT_BAL_ENV, false);
+        }
+
+        while (argIndex < callIns.args.size()) {
+            BIROperand arg = callIns.args.get(argIndex);
+            this.loadVar(arg.variableDcl);
+            argIndex += 1;
+        }
+        String jClassName = callIns.jClassName;
+        String jMethodName = callIns.name;
+        this.mv.visitMethodInsn(callIns.invocationType, jClassName, jMethodName, jMethodVMSig, isInterface);
```

Comment:
```suggestion
        this.mv.visitMethodInsn(callIns.invocationType, callIns.jClassName, callIns.name, jMethodVMSig, isInterface);
```
As we use the `jClassName` and `name` fields only once, we can use them directly instead of creating variables.